internal/objects/reconciler_test.go

```diff
@@ -215,11 +215,12 @@ var _ = Describe("Exceptions", func() {
 				}
 
 				// update the role with the selector and treeSelector annotation
-				UpdateObjectWithAnnotations(ctx, api.RoleResource, names[p], "testrole", map[string]string{
+				err := UpdateObjectWithAnnotations(ctx, api.RoleResource, names[p], "testrole", map[string]string{
```

Comment:
Are there any cases today where `UpdateObjectWithAnnotations` is _supposed_ to fail? If not, maybe put the `Expect(...).Should(Succeed())` right inside the function?